cmd/thanos/sidecar.go

```diff
@@ -294,6 +299,23 @@ func runSidecar(
 	return nil
 }
 
+func checkFlags(ctx context.Context, logger log.Logger, promURL *url.URL, tsdbPath string) error {
+	flags, err := promclient.ConfiguredFlags(ctx, logger, promURL)
+	if err != nil {
+		return errors.Wrap(err, "configured flags; failed to check flags")
+	}
+	// Check if sidecar has the same tsdbpath with prometheus
+	if flags.TSDBPath != tsdbPath {
+		return errors.New("TSDB path mismatches with prometheus")
+	}
+	// Check if min-block-time and max-block-time are the same
+	if flags.TSDBMinTime != flags.TSDBMaxTime {
```

Comment:
Let's check also if it equals `2h` !

We may want to produce warning only if it's not? To allow this as debug thing only.

Reply:
Thanks for your review! I' m new to thanos so just a small question. Why it should be 2h for block-time?

Reply:
Exactly. TSDB is designed for 2h only.